scripts/run-node.py

```diff
@@ -1,119 +1,176 @@
-import os
+#!/usr/bin/env python3
+
 import subprocess
-import json
+import sys
+import os
 import requests
+import json
 import zipfile
-
 from io import BytesIO
 
-# Mapping of env to chain_id
-ENV_TO_CHAIN_ID = {
-    "local": None,
-    "devnet": "arctic-1",
-    "testnet": "atlantic-2"
+# req packages
+dependencies = ['requests']
+
+for package in dependencies:
+    try:
+        __import__(package)
+    except ImportError:
+        print(f"Installing the '{package}' package...")
+        subprocess.check_call([sys.executable, "-m", "pip", "install", package])
+
+# advanced user configs
+moniker = "pynode" # optional custom moniker for the node
+trust_height_delta = 40000  # negative height offset for state sync
+enable_unsafe_reset = True  # wipe database and keys before setup
+version_override = False  # override version fetching. if true, specify version(s) below
+
+# chain binary version ["version_override" must be true to use]
+MAINNET_VERSION = "v3.9.0"
+DEVNET_VERSION = "v5.2.2"
+TESTNET_VERSION = "v5.2.2"
+
+# map env to chain ID and optional manual version override
+ENV_TO_CONFIG = {
+    "local": {"chain_id": None, "version": "latest"},
+    "devnet": {"chain_id": "arctic-1", "version": DEVNET_VERSION},
+    "testnet": {"chain_id": "atlantic-2", "version": TESTNET_VERSION},
+    "mainnet": {"chain_id": "pacific-1", "version": MAINNET_VERSION}
```

Comment:
now that we're fetching from abci_info we could remove this (and probably should? to avoid app-hashes)

Reply:
I left that in there for manual override incase someone wants a specific version for whatever reason, testing or debug etc